src/memex/presenters.py

```diff
@@ -6,10 +6,13 @@
 import collections
 import copy
 
+from pyramid import security
+
 
 class AnnotationBasePresenter(object):
-    def __init__(self, annotation, links_service):
-        self.annotation = annotation
+    def __init__(self, annotation_resource, links_service):
```

Comment:
I'm not sure.

It doesn't do anything with the group service now. But it would do if you moved the creation of the resource object into it.

At the moment I think it's cleaner for the presenter to have a small set of services it depends on, otherwise you end up in weird situations like the one you have in the `update_annotation` view where you have a context object which is an `AnnotationResource`, but it's the wrong one, so you now have to create a new one just to pass it into the presenter.